// @medusajs/framework@2.12.2/dist/http/middlewares/apply-locale.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.applyLocale = applyLocale;
const utils_1 = require("@medusajs/utils");
const CONTENT_LANGUAGE_HEADER = "content-language";
/**
 * Middleware that resolves the locale for the current request.
 *
 * Resolution order:
 * 1. Query parameter `?locale=en-US`
 * 2. Content-Language header
 *
 * The resolved locale is set on `req.locale`.
 */
async function applyLocale(req, _, next) {
    // 1. Check query parameter
    const queryLocale = req.query.locale;
    if (queryLocale) {
        req.locale = (0, utils_1.normalizeLocale)(queryLocale);
        return next();
    }
    // 2. Check Content-Language header
    const headerLocale = req.get(CONTENT_LANGUAGE_HEADER);
    if (headerLocale) {
        req.locale = (0, utils_1.normalizeLocale)(headerLocale);
        return next();
    }
    const query = req.scope.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const { data: [store], } = await query.graph({
        entity: "store",
        fields: ["id", "supported_locales"],
        pagination: {
            take: 1,
        },
    }, {
        cache: {
            enable: true,
        },
    });
    if (store?.supported_locales?.length) {
        req.locale = store.supported_locales.find((locale) => locale.is_default)?.locale_code;
        return next();
    }
    return next();
}
